pkg/reconciler/apis/apibindingdeletion/apibinding_deletor_test.go

```diff
@@ -163,34 +155,21 @@ func TestAPIBindingTerminating(t *testing.T) {
 
 	tests := []struct {
 		name                      string
-		existingObject            []runtime.Object
-		metadataClientActionSet   metaActionSet
-		expectErrorOnDelete       error
+		existingObjects           map[schema.GroupVersionResource]*metav1.PartialObjectMetadata
 		expectedResourceRemaining gvrDeletionMetadataTotal
 	}{
 		{
-			name:           "no resource left for apibinding to delete",
-			existingObject: []runtime.Object{},
-			metadataClientActionSet: []metaAction{
-				{"pods", "list"},
-				{"deployments", "list"},
-			},
+			name: "no resource left for apibinding to delete",
 			expectedResourceRemaining: gvrDeletionMetadataTotal{
 				gvrToNumRemaining:        map[schema.GroupVersionResource]int{},
 				finalizersToNumRemaining: map[string]int{},
 			},
 		},
 		{
 			name: "some resource remaining after apibinding deletion",
-			existingObject: []runtime.Object{
-				newPartialObject("v1", "Pod", "pod1", "ns1", nil),
-				newPartialObject("apps/v1", "Deployment", "deploy1", "ns1", nil),
-			},
-			metadataClientActionSet: []metaAction{
```

Comment:
I removed any check for the client calls here since it seems sufficient to validate the method we're testing - just inputs and outputs - without assuming implementation details in there.

Reply:
I kinda think we should be validating what it's issuing - e.g. at least that it's trying to delete the right stuff

Reply:
Happy to have someone do that in a follow-up - notably, the original test did not do that.

Reply:
File an issue or add a TODO?